### PR TITLE
My code review changes.

### PR DESCRIPTION
**Pull request by beaslecd**
### Description of Project ### 
This application allows users to estimate the cost of shipping an item from one destination to another. The user enters all necessary information for the package and its source. With the information, the programs return an estimated shipping cost for the item with various shipping vendors. 

The program was available on time.

The program is easy for me to understand. Variable names are clear, methods are properly named, JavaDoc provides all necessary information.

The program did compile. 

### Three Technical Suggestions ###
- To follow the OOP concept of Encapsulation, add 'private' to all the attributes within the DTO classes. That way those attributes are hidden and can only be changed with the set method and called with get method.
- Add return description in the JavaDoc to allow you or whoever is reading the code to figure out what exactly is being returned so there's no or little confusion.
- For the drop down lists, I would make the default selection a carrier instead of a blank selection so that the user would get a better understanding.

### What I learned ###
- Bootstrap really helps with making dynamic HTML pages
- Comments and JavaDoc helps with understanding the application.
- I learned more about endpoints and JSON as they provide really good information for an application.

### CODE REVIEW

File: src/main/java/com/shippingcostestimator/enterprise/PackageEstimatorController.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
     /**
      * Handle the root ("/") endpoint and return a start page.
-     * @return
+     * @return start.html page.
      */
```
Comment (beaslecd) [inline src/main/java/com/shippingcostestimator/enterprise/PackageEstimatorController.java:24]: Added return description for JavaDoc. [Ref](https://medium.com/@rhamedy/a-short-summary-of-java-coding-best-practices-31283d0167d3)

File: src/main/java/com/shippingcostestimator/enterprise/PackageEstimatorController.java [hunk 2]
```diff
@@ -39,7 +39,7 @@ public String index() {
     /***
      * Handle the saveEstimate ("/saveEstimate") endpoint and save a Shipment object using data specified by the user.
      * @param shipment
-     * @return
+     * @return start.html page.
      */
```
Comment (beaslecd) [inline src/main/java/com/shippingcostestimator/enterprise/PackageEstimatorController.java:43]: Added return description for JavaDoc. [Ref](https://medium.com/@rhamedy/a-short-summary-of-java-coding-best-practices-31283d0167d3)

File: src/main/java/com/shippingcostestimator/enterprise/PackageEstimatorController.java [hunk 3]
```diff
@@ -55,12 +55,11 @@ public String saveEstimate(Shipment shipment){
 
     /***
      * Handle the shipment ("/shipment") endpoint and return all Shipment objects that have been created from data specified by the user.
-     * @return
+     * @return all shipments in the list.
      */
```
Comment (beaslecd) [inline src/main/java/com/shippingcostestimator/enterprise/PackageEstimatorController.java:59]: Added return description for JavaDoc. [Ref](https://medium.com/@rhamedy/a-short-summary-of-java-coding-best-practices-31283d0167d3)

File: src/main/java/com/shippingcostestimator/enterprise/PackageEstimatorController.java [hunk 4]
```diff
@@ -55,12 +55,11 @@ public String saveEstimate(Shipment shipment){
...
     @GetMapping("/shipment")
     @ResponseBody
     public List<Shipment> fetchAllShipments(){
-
         return shipmentService.fetchAllShipments();
```
Comment (beaslecd) [inline src/main/java/com/shippingcostestimator/enterprise/PackageEstimatorController.java:64]: Fixed format for method body for the fetchAllShipments method.  Makes it easier to read. [Ref](https://medium.com/@rhamedy/a-short-summary-of-java-coding-best-practices-31283d0167d3)

File: src/main/java/com/shippingcostestimator/enterprise/dao/ShipmentDAOStub.java [hunk 5]
```diff
@@ -17,7 +17,7 @@
     /***
      * Returns a Shipment object with a specified integer ID.
      * @param id
-     * @return
+     * @return  a specific shipment based on shipmentId.
      */
```
Comment (beaslecd) [inline src/main/java/com/shippingcostestimator/enterprise/dao/ShipmentDAOStub.java:21]: Added return description JavaDoc. [Ref](https://medium.com/@rhamedy/a-short-summary-of-java-coding-best-practices-31283d0167d3)

File: src/main/java/com/shippingcostestimator/enterprise/dao/ShipmentDAOStub.java [hunk 6]
```diff
@@ -27,7 +27,7 @@ public Shipment findShipId(int id) {
     /***
      * Saves a Shipment estimate.
      * @param shipment
-     * @return
+     * @return Saved shipment estimate.
      */
```
Comment (beaslecd) [inline src/main/java/com/shippingcostestimator/enterprise/dao/ShipmentDAOStub.java:31]: Added return description  JavaDoc. [Ref](https://medium.com/@rhamedy/a-short-summary-of-java-coding-best-practices-31283d0167d3)

File: src/main/java/com/shippingcostestimator/enterprise/dao/ShipmentDAOStub.java [hunk 7]
```diff
@@ -38,7 +38,7 @@ public Shipment saveEstimate(Shipment shipment) {
 
     /***
      * Returns a list of Shipment objects that are currently in use by the program.
-     * @return
+     * @return A list of shipments.
      */
```
Comment (beaslecd) [inline src/main/java/com/shippingcostestimator/enterprise/dao/ShipmentDAOStub.java:42]: Added return description JavaDoc. [Ref](https://medium.com/@rhamedy/a-short-summary-of-java-coding-best-practices-31283d0167d3)

File: src/main/java/com/shippingcostestimator/enterprise/dto/Destination.java [hunk 8]
```diff
@@ -12,6 +12,6 @@
  */
 public @Data
 class Destination {
-    String startAddress;
-    String endAddress;
+    private String startAddress;
+    private String endAddress;
```
Comment (beaslecd) [inline src/main/java/com/shippingcostestimator/enterprise/dto/Destination.java:16]: Added 'private' to the attributes so it'll follow the encapsulation concept for Oobject Oriented Programming. [Ref](https://www.geeksforgeeks.org/encapsulation-in-java/)

File: src/main/java/com/shippingcostestimator/enterprise/dto/ShippingCost.java [hunk 9]
```diff
@@ -14,8 +14,8 @@
...
 public @Data
 class ShippingCost {
-    int totalCostId;
-    int serviceLevel;
-    int rates;
-    int estArrival;
+    private int totalCostId;
+    private int serviceLevel;
+    private int rates;
+    private int estArrival;
```
Comment (beaslecd) [inline src/main/java/com/shippingcostestimator/enterprise/dto/ShippingCost.java:20]: Added 'private' to the attributes so it'll follow the encapsulation concept for Object-Oriented Programming [Ref](https://www.geeksforgeeks.org/encapsulation-in-java/)

File: src/main/java/com/shippingcostestimator/enterprise/service/ShipmentServiceStub.java [hunk 10]
```diff
@@ -31,7 +31,7 @@ public ShipmentServiceStub(IShipmentDAO shipmentDAO){
     /***
      * Returns a Shipment object with a specified integer ID.
      * @param id
-     * @return
+     * @return A shipment estimate based on specified ShipmentId.
      */
```
Comment (beaslecd) [inline src/main/java/com/shippingcostestimator/enterprise/service/ShipmentServiceStub.java:35]: Added return description JavaDoc [Ref.](https://medium.com/@rhamedy/a-short-summary-of-java-coding-best-practices-31283d0167d3)

File: src/main/java/com/shippingcostestimator/enterprise/service/ShipmentServiceStub.java [hunk 11]
```diff
@@ -42,7 +42,7 @@ public Shipment findShipmentId(int id) {
     /***
      * Saves a Shipment estimate.
      * @param shipment
-     * @return
+     * @return Saved shipment estimate.
```
Comment (beaslecd) [inline src/main/java/com/shippingcostestimator/enterprise/service/ShipmentServiceStub.java:45]: Added return description JavaDoc. [Ref](https://medium.com/@rhamedy/a-short-summary-of-java-coding-best-practices-31283d0167d3)

File: src/main/java/com/shippingcostestimator/enterprise/service/ShipmentServiceStub.java [hunk 12]
```diff
@@ -52,7 +52,7 @@ public Shipment saveEstimate(Shipment shipment) throws Exception {
 
     /***
      * Returns a list of all Shipment objects that are currently in use by the program.
-     * @return
+     * @return a list of all shipments.
```
Comment (beaslecd) [inline src/main/java/com/shippingcostestimator/enterprise/service/ShipmentServiceStub.java:55]: Added return description JavaDoc. [Ref.](https://medium.com/@rhamedy/a-short-summary-of-java-coding-best-practices-31283d0167d3)

File: src/main/resources/templates/start.html [hunk 13]
```diff
@@ -1,5 +1,5 @@
 <!DOCTYPE html>
-<html xmls:th="http://www.thymeleaf.org">
+<html xmlns:th="http://www.thymeleaf.org">
```
Comment (beaslecd) [inline src/main/resources/templates/start.html:2]: Fixed typo in the HTML thymeleaf tag.

**Review by beaslecd — COMMENTED** [review]
[Ref.](https://medium.com/@rhamedy/a-short-summary-of-java-coding-best-practices-31283d0167d3)

File: src/main/java/com/shippingcostestimator/enterprise/dao/ShipmentDAOStub.java (hunk 5 above)
Comment (discospiff) [reply src/main/java/com/shippingcostestimator/enterprise/dao/ShipmentDAOStub.java:21]: While you're here, describe the parameters as well, in the @param tag.

File: src/main/java/com/shippingcostestimator/enterprise/dto/Destination.java (hunk 8 above)
Comment (discospiff) [reply src/main/java/com/shippingcostestimator/enterprise/dto/Destination.java:16]: Good.

File: src/main/resources/templates/start.html (hunk 13 above)
Comment (discospiff) [reply src/main/resources/templates/start.html:2]: Good find!

**Comment by discospiff** [comment]
Several good suggestions here, which decrease technical debt.  I recommend merging these changes - and then enhance the JavaDoc.

**Pull request merged**